src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,88 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useEffect } from 'react';
+import { Todo } from '../../types/Todo';
+import { SortType } from '../../types/sortType';
+
+interface Props {
+  todos: Todo[];
+  query: string;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  setSortType: React.Dispatch<React.SetStateAction<SortType>>;
+}
+
+export const TodoFilter: React.FC<Props> = ({
+  todos,
+  setTodos,
+  query,
+  setQuery,
+  setSortType,
+}) => {
+  useEffect(() => {
+    setTodos(todos);
+  }, [query]);
```

Comment:
I dont think you should do it here. So passing todos and setTodos to this component via props is not really nevessary